### PR TITLE
docs(diffusers): Docs API part(loaders/models/pipelines/internal_class)

### PR DESCRIPTION
**Pull request by wcrzlh**
# What does this PR do?

This PR is proposed to add documentation of diffusers (API-loaders/models/pipelines/internal_class).

Feature:

- [x] diffusers documentation(API-loaders/models/pipelines/internal_class)

BugFix:

- [x] fix training weights cannot be loaded
- [x] fix dtype mismatch in groupnorm on training

## Before submitting
- [ ] This PR fixes a typo or improves the docs (you can dismiss the other checks if that's the case).
- [ ] Did you read the [contributor guideline](https://github.com/mindspore-lab/mindone/blob/master/CONTRIBUTING.md)?
- [ ] Did you make sure to update the documentation with your changes? E.g. record bug fixes or new features in `What's New`. Here are the
      [documentation guidelines](https://github.com/mindspore-lab/mindcv/wiki/%E6%96%87%E6%A1%A3%E7%BC%96%E5%86%99%E8%A7%84%E8%8C%83)
- [ ] Did you build and run the code without any errors?
- [ ] Did you report the running environment (NPU type/MS version) and performance in the doc? (better record it for data loading, model inference, or training tasks)
- [ ] Did you write any new necessary tests?

## Who can review?

Anyone in the community is free to review the PR once the tests have passed. Feel free to tag
members/contributors who may be interested in your PR.

@xxx

### CODE REVIEW

**Review by CaitinZhao — APPROVED** [review]

File: docs/diffusers/api/loaders/lora.md [hunk 1]
```diff
@@ -0,0 +1,27 @@
...
+# LoRA
+
+LoRA is a fast and lightweight training method that inserts and trains a significantly smaller number of parameters instead of all the model parameters. This produces a smaller file (~100 MBs) and makes it easier to quickly train a model to learn a new concept. LoRA weights are typically loaded into the UNet, text encoder or both. There are two classes for loading LoRA weights:
+
+- `LoraLoaderMixin` provides functions for loading and unloading, fusing and unfusing, enabling and disabling, and more functions for managing LoRA weights. This class can be used with any model.
+- `StableDiffusionXLLoraLoaderMixin` is a [Stable Diffusion (SDXL)](../../api/pipelines/stable_diffusion/stable_diffusion_xl.md) version of the `LoraLoaderMixin` class for loading and saving LoRA weights. It can only be used with the SDXL model.
+
+!!! tip
+
+    To learn more about how to load LoRA weights, see the [LoRA](../../using-diffusers/loading_adapters.md#lora) loading guide.
```
Comment (SamitHuang) [inline docs/diffusers/api/loaders/lora.md:22]: please fix the display issue on the hyper-link

File: docs/diffusers/api/loaders/peft.md [hunk 2]
```diff
@@ -0,0 +1,21 @@
...
+specific language governing permissions and limitations under the License.
+-->
+
+# PEFT
+
+Diffusers supports loading adapters such as [LoRA](../../using-diffusers/loading_adapters.md) with the [PEFT](https://huggingface.co/docs/peft/index) library with the [`loaders.peft.PeftAdapterMixin`](peft.md#mindone.diffusers.loaders.peft.PeftAdapterMixin) class. This allows modeling classes in Diffusers like [`UNet2DConditionModel`](../models/unet2d-cond.md#unet2dconditionmodel) to load an adapter.
+
+!!! tip
+
+    Refer to the [Inference with PEFT](../../tutorials/using_peft_for_inference.md) tutorial for an overview of how to use PEFT in Diffusers for inference.
```
Comment (SamitHuang) [inline docs/diffusers/api/loaders/peft.md:19]: same here, fix

File: docs/diffusers/api/loaders/single_file.md [hunk 3]
```diff
@@ -0,0 +1,52 @@
...
+# Single files
+
+The [`from_single_file`](single_file.md#mindone.diffusers.loaders.single_file.FromSingleFileMixin) method allows you to load:
+
+* a model stored in a single file, which is useful if you're working with models from the diffusion ecosystem, like Automatic1111, and commonly rely on a single-file layout to store and share models
+* a model stored in their originally distributed layout, which is useful if you're working with models finetuned with other services, and want to load it directly into Diffusers model objects and pipelines
+
+!!! tip
+
+    Read the [Model files and layouts](../../using-diffusers/other-formats.md) guide to learn more about the Diffusers-multifolder layout versus the single-file layout, and how to load models stored in these different layouts.
```
Comment (SamitHuang) [inline docs/diffusers/api/loaders/single_file.md:22]: same here, fix

File: docs/diffusers/api/loaders/unet.md [hunk 4]
```diff
@@ -0,0 +1,23 @@
...
+
+# UNet
+
+Some training methods - like LoRA and Custom Diffusion - typically target the UNet's attention layers, but these training methods can also target other non-attention layers. Instead of training all of a model's parameters, only a subset of the parameters are trained, which is faster and more efficient. This class is useful if you're *only* loading weights into a UNet. If you need to load weights into the text encoder or a text encoder and UNet, try using the [`load_lora_weights`](lora.md#mindone.diffusers.loaders.lora.LoraLoaderMixin.load_lora_weights) function instead.
+
+The [`UNet2DConditionLoadersMixin`](unet.md#mindone.diffusers.loaders.unet.UNet2DConditionLoadersMixin) class provides functions for loading and saving weights, fusing and unfusing LoRAs, disabling and enabling LoRAs, and setting and deleting adapters.
+
+!!! tip
+
+    To learn more about how to load LoRA weights, see the [LoRA](../../using-diffusers/loading_adapters.md#lora) loading guide.
```
Comment (SamitHuang) [inline docs/diffusers/api/loaders/unet.md:21]: same here

**Review by SamitHuang — COMMENTED** [review]
!!! tip ... hyperlink

use markdown warning symbol, and fix the hyperlink display

File: docs/diffusers/api/loaders/lora.md (hunk 1 above)
Comment (SamitHuang) [reply docs/diffusers/api/loaders/lora.md:22]: > [!tip]
> To learn more about how to load LoRA weights, see the [LoRA](../../using-diffusers/loading_adapters.md#lora) loading guide.

Comment (wcrzlh) [reply docs/diffusers/api/loaders/lora.md:22]: The hyper-link directs to PR(https://github.com/mindspore-lab/mindone/pull/679). After merging both pr, the hyper-link could be displayed correctly.

**Review by vigo999 — APPROVED** [review]

**Pull request merged**